universal-application-tool-0.0.1/app/views/admin/programs/ProgramBlockEditView.java

```diff
@@ -374,49 +377,55 @@ private ContainerTag renderQuestion(
             .with(p(questionDefinition.getName()))
             .with(p(questionDefinition.getDescription()).withClasses(Styles.MT_1, Styles.TEXT_SM));
 
-    ContainerTag optionalButton =
-        TagCreator.button()
-            .withClasses(
-                Styles.FLEX,
-                Styles.GAP_2,
-                Styles.ITEMS_CENTER,
-                isOptional ? Styles.TEXT_BLACK : Styles.TEXT_GRAY_400,
-                Styles.FONT_MEDIUM,
-                Styles.BG_TRANSPARENT,
-                Styles.ROUNDED_FULL,
-                StyleUtils.hover(Styles.BG_GRAY_400, Styles.TEXT_GRAY_300))
-            .withType("submit")
-            .with(p("optional").withClasses("hover-group:text-white"))
-            .with(
-                div()
-                    .withClasses(Styles.RELATIVE)
-                    .with(
-                        div()
-                            .withClasses(
-                                isOptional ? Styles.BG_BLUE_600 : Styles.BG_GRAY_600,
-                                Styles.W_14,
-                                Styles.H_8,
-                                Styles.ROUNDED_FULL))
-                    .with(
-                        div()
-                            .withClasses(
-                                Styles.ABSOLUTE,
-                                Styles.BG_WHITE,
-                                isOptional ? Styles.RIGHT_1 : Styles.LEFT_1,
-                                Styles.TOP_1,
-                                Styles.W_6,
-                                Styles.H_6,
-                                Styles.ROUNDED_FULL)));
-    String toggleOptionalAction =
-        controllers.admin.routes.AdminProgramBlockQuestionsController.setOptional(
-                programDefinitionId, blockDefinitionId, questionDefinition.getId())
-            .url();
-    ContainerTag optionalToggle =
-        form(csrfTag)
-            .withMethod(HttpVerbs.POST)
-            .withAction(toggleOptionalAction)
-            .with(input().isHidden().withName("optional").withValue(isOptional ? "false" : "true"))
-            .with(optionalButton);
+    Tag optionalToggle;
```

Comment:
Suggestion (non-blocking): extract the optional toggle logic to a private method and use an early return if it shouldn't be rendered

Reply:
done